server/controller/semester.controller.js

```diff
@@ -165,40 +154,40 @@ semesterController.delete(
   '/',
   passport.authenticate('jwt', { session: false }),
   validateSemesterId,
-  async (req, res) => {
-    console.log('Request ', req.query);
-    const errorsAfterValidation = validationResult(req);
-    if (!errorsAfterValidation.isEmpty()) {
-      res.status(400).json({
-        code: 400,
-        errors: errorsAfterValidation.mapped(),
-      });
-    } else {
+  (req, res) => {
+    validationHandler(req, res, () => {
       try {
-        // Get semester ObjectId from query
-        const semester_id = req.query.semester_id;
-        const deletedSemester = await Semester.findByIdAndDelete({
-          _id: semester_id,
-        });
-
-        if (deletedSemester) {
-          return res
-            .status(200)
-            .json({ message: 'deleted a semester', deletedSemester });
-        } else {
-          generateServerErrorCode(
-            res,
-            403,
-            'delete semester error',
-            SEMESTER_NOT_FOUND,
-            'semester_id'
-          );
-        }
+        const { _id } = req.query;
+        Semester.findByIdAndDelete(
+          {
+            _id,
+          },
+          (err, deletedSemester) => {
+            if (deletedSemester)
+              return res
+                .status(200)
+                .json({ message: 'deleted a semester: ', deletedSemester });
+            return generateServerErrorCode(
+              res,
+              403,
+              'delete semester error',
+              SEMESTER_NOT_FOUND,
+              'semester_id'
+            );
+          }
+        );
       } catch (e) {
         generateServerErrorCode(res, 500, e, SOME_THING_WENT_WRONG);
       }
-    }
+    });
   }
 );
 
+semesterController.delete('/all', (req, res) => {
```

Comment:
Same comment as in course controller: Add authentication. Though eventually this would only be authorized for just the admin to delete an entire collection.